frontend/lib/resolution/Resolver.cpp

```diff
@@ -1893,32 +1904,70 @@ void Resolver::resolveTupleDecl(const TupleDecl* td,
   QualifiedType::Kind declKind = (Qualifier) td->intentOrKind();
   QualifiedType useT;
 
+  // Non-default intents are currently not allowed for tuple-grouped formals.
+  //
+  // Note: This doesn't apply to nested TupleDecls like ``(a, b)`` in the
+  // formal ``((a, b), c)``. Such nested formals should be handled by
+  // ``resolveTupleUnpackDecl``.
+  if (td->isTupleDeclFormal() && declKind != QualifiedType::DEFAULT_INTENT) {
+    useT = QualifiedType(declKind, ErroneousType::get(context));
+    ResolvedExpression& result = byPostorder.byAst(td);
+    result.setType(useT);
+    return;
+  }
+
   // Figure out the type to use for this tuple
   if (useType != nullptr) {
     useT = QualifiedType(declKind, useType);
+  } else if (substitutions != nullptr &&
+             substitutions->count(td->id()) != 0) {
+    // TODO: should this be referential or value?
```

Comment:
I say "referential" because of [this spec language](https://chapel-lang.org/docs/language/spec/tuples.html#value-tuples-and-referential-tuples):

> A referential tuple is analogous to a group of formal arguments that each have the default argument intent. Referential tuples include:
> - tuple expressions
> - formal arguments with the default or const argument intent that have a tuple type

(And a little bit above there it says tuple formals with `in` intent are value tuples, though that doesn't matter at this point).

Reply:
Yup, you're right! The new code below handles this, but I forgot to update the comment.